target/linux/ath79/generic/base-files/etc/board.d/02_network

```diff
@@ -239,6 +239,11 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "2:lan:3" "3:lan:2" "4:lan:1" "1:wan"
 		;;
+	tplink,archer-d7-v1|\
+	tplink,archer-d7b-v1)
+		ucidef_add_switch "switch0" \
+			"0@eth1" "3:lan:3" "4:lan:2" "5:lan:1" "6@eth0" "2:wan:4" "1:wan:5"
```

Comment:
In the commit message you say eth0 is not working ("WIP"). Can you provide more details on that? So, is just DSL not working, and this is acting as a switch, or should we better disable the ports?

Reply:
Despite, you use fixed-link for eth1 in DTS, so how can you actually have two wan-ports then?

Reply:
One port of the cpu is connected to the switch and the other port of the cpu is connected to the Broadcom cpu.
The last port of the switch is labeled as LAN/WAN so i chose to use this as wan port.

For what concerns the DSL part, the configuration of the DSL part is stored on the BCM flash. 
If the dsl link is active and the user configured the modem using the OEM fw, it could work. However I don't have an adsl connection anymore so I can't test it but I chose to left it enabled because it doesn't cause trouble (many users alreay tested it https://forum.openwrt.org/t/archer-d7-support/15790/118)

I didn't get the question about the fixed-link... could you explain it? (I am sorry but I have very little experience in the dts world :disappointed: )

Reply:
> The last port of the switch is labeled as LAN/WAN so i chose to use this as wan port.

So, this is either 2: or 1: (internally). Which is it and what's the other one then.

Reply:
"2:wan:4" is the wan on the back of the router and it's labeled as WAN1 in the switch configuration page
"1:wan:5" is the internal wan and it's labeled as WAN2 in the switch configuration page

![image](https://user-images.githubusercontent.com/2470733/74292470-636fcc80-4d38-11ea-9731-74f5b4637600.png)

Reply:
So, that means you configure both WAN1 and WAN2 as if connected to eth0.

Reply:
Yes.